src/software/thunderscope/robot_communication.py

```diff
@@ -111,41 +154,45 @@ def run(self):
                     ),
                 )
 
-                primitive_set = PrimitiveSet(
-                    time_sent=Timestamp(epoch_timestamp_seconds=time.time()),
-                    stay_away_from_ball=False,
-                    robot_primitives={
-                        robot_id: Primitive(direct_control=diagnostics_primitive)
-                        for robot_id in self.robots_connected_to_diagnostics
-                    },
-                    sequence_number=self.sequence_number,
+                # for all robots connected to diagnostics, set their primitive
+                for robot_id in self.robots_connected_to_diagnostics:
+                    robot_primitives[robot_id] = Primitive(
+                        direct_control=diagnostics_primitive
+                    )
+
+            # sends a final stop primitive to all disconnected robots and removes them from list
+            # in order to prevent robots acting on cached old primitives
+            while self.robots_to_be_disconnected:
+                robot_primitives[self.robots_to_be_disconnected.pop()] = Primitive(
+                    stop=StopPrimitive()
                 )
 
-                self.sequence_number += 1
-
-                if self.estop_reader.isEstopPlay():
-                    self.last_time = primitive_set.time_sent.epoch_timestamp_seconds
-                    self.send_primitive_set.send_proto(primitive_set)
-
-                time.sleep(0.001)
-
-    def connect_fullsystem_to_robots(self):
-        """ Connect the robots to fullsystem """
-
-        self.fullsystem_connected_to_robots = True
-        self.robots_connected_to_handheld_controllers = set()
-        self.robots_connected_to_diagnostics = set()
+            # initialize total primitive set and send it
+            primitive_set = PrimitiveSet(
+                time_sent=Timestamp(epoch_timestamp_seconds=time.time()),
+                stay_away_from_ball=False,
+                robot_primitives=robot_primitives,
```

Comment:
Not sure if it's expected behaviour, but I get the following error when I test locally with `--run_blue --run_diagnositcs` flags: 
```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/home/nima/.cache/bazel/_bazel_nima/d4d8c6a2a61c53de6e279fde073469b4/execroot/__main__/bazel-out/k8-fastbuild/bin/software/thunderscope/thunderscope_main.runfiles/__main__/software/thunderscope/robot_communication.py", line 181, in run
    primitive_set = PrimitiveSet(
TypeError: Argument robot_primitives is not iterable
```
Seems like in my case, robot_primitives is an empty primitive set of type `proto.tbots_software_msgs_pb2.PrimitiveSet` which I guess is not an iterable!?